lib/python/qmk/cli/lint.py

```diff
@@ -46,6 +48,11 @@ def lint(cli):
         ok = False
         cli.log.error('Missing %s', readme_path)
 
+    # Check keyboard name is valid
+    if not re.fullmatch(r"[a-z0-9][a-z0-9_]*", cli.config.lint.keyboard):
+        ok = False
+        cli.log.error('Invalid keyboard name')
```

Comment:
```suggestion
        cli.log.error('Invalid keyboard name! Guidelines: https://docs.qmk.fm/#/hardware_keyboard_guidelines?id=naming-your-keyboardproject')
```
Providing the link in the output might be helpful.

Reply:
This isn't a pattern that's used outside doctor right now. For example, `info.json` doesn't point to <https://docs.qmk.fm/#/reference_info_json>. Is this a general strategy we want to move towards?